### PR TITLE
pull mapping_stats.py meta data from mapping_stats.json

### PR DESCRIPTION
**Pull request by ben-lerch**
this is first draft of the effort. deeper revisions could be made eventually.

### CODE REVIEW

File: pbreports/report/mapping_stats.py [hunk 1]
```diff
@@ -658,49 +674,22 @@ class MappingStatsCollector(object):
...
-        Constants.A_READLENGTH_MAX: "The maximum length of polymerase reads that mapped to the reference sequence",
-        Constants.A_SUBREAD_LENGTH_MAX: "The maximum length of subreads that mapped to the reference sequence",
-        Constants.A_SUBREAD_LENGTH_N50: "The subread length at which 50% of the mapped bases are in subreads longer than, or equal to, this value",
-        Constants.A_READLENGTH_N50: "The read length at which 50% of the mapped bases are in polymerase reads longer than, or equal to, this value",
-    }
+
+    ATTR_LABELS_LIST = []
+    for id in meta_rpt._attr_dict.keys():
+    	ATTR_LABELS_LIST.append((id, meta_rpt.get_meta_attribute(id).name))
+    ATTR_LABELS = OrderedDict(ATTR_LABELS_LIST)
```
Comment (natechols) [inline pbreports/report/mapping_stats.py]: I don't think we actually need to keep ATTR_LABELS, we can just collect the labels when we need them.

**Comment by ben-lerch** [comment]
it gets used here:
attributes = get_attributes(_total_aggregators, self.ATTR_LABELS)
it's not clear to me how to remove that line, but to_report could probably be improved more generally using the report_spec.py machinery. for a lightweight revision of mapping_stats.py i'm not sure how to get rid of ATTR_LABELS.

**Comment by natechols** [comment]
Just change the `get_attributes` function to pull labels from meta_rpt instead of passing the separate dict.

**Comment by ben-lerch** [comment]
ok done. i had actually written this version of get_attributes before but i thought it didn't work- i guess it was something else that failed but threw an error message referencing get_attributes.

**Pull request merged**